File: 1. Two Sum/twoSum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target) {
    int *ret = (int*) calloc( 2, sizeof(int));
    int x;
    int y;
    int delta;
    
    for( x=1; x<numsSize; x++ )
    {
        delta = target-nums[x];

        for( int y=0; y<x; y++ )
        {
            if( nums[y]==delta )
            {
                ret[0] = y;
                ret[1] = x;
                return ret;
            }
        }
    }
    
    return ret;
}

File: 1. Two Sum/twoSum.go
func twoSum(nums []int, target int) []int {
    var x int
    var y int
    var delta int
    for x=1; x<len(nums); x++ {
        delta = target - nums[x]
        
        for y=0; y<x; y++ {
            if delta==nums[y] {
                return []int { y, x}
            }
        }
    }
    
    return []int {0,0}
}